src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -470,10 +470,17 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
             image_data_disks = [{'lun': disk.lun} for disk in image_data_disks]
 
         elif res['type'].lower() == 'galleries':
-            image_info = compute_client.gallery_images.get(resource_group_name=res['resource_group'],
-                                                           gallery_name=res['name'],
-                                                           gallery_image_name=res['child_name_1'])
-            namespace.os_type = image_info.os_type
+            try:
+                image_info = compute_client.gallery_images.get(
+                    resource_group_name=res['resource_group'],
+                    gallery_name=res['name'],
+                    gallery_image_name=res['child_name_1'])
+                namespace.os_type = image_info.os_type
+            except Exception:  # pylint: disable=broad-except
+                logger.warning('Failed to get shared gallery image definition.')
+                if not namespace.os_type:
+                    logger.warning('OS type is not specified. Assume it is Linux. You can use --os-type to specify it.')
+                    namespace.os_type = 'Linux'
```

Comment:
```suggestion
            except ValidationError:
                logger.debug('Failed to get shared gallery image definition.')
```

Reply:
according to https://github.com/Azure/azure-cli/blob/8c04af8c0bd508b00c28a21178a6e713060fbbb6/src/azure-cli/azure/cli/command_modules/vm/_validators.py#L407-L410

`--os-type` is forbidden for shared gallery image, you will get the following error if specifying both:
```
invalid usage for storage profile: create managed OS disk from custom image:
        not applicable: --os-type
```

---

I think `namespace.os_type` could be skipped here, the default will be set later in
https://github.com/Azure/azure-cli/blob/8c04af8c0bd508b00c28a21178a6e713060fbbb6/src/azure-cli/azure/cli/command_modules/vm/_validators.py#L508-L509

Reply:
@abuccts Thank you for your investigation. I updated the PR.